<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Tab_Opt</class>
 <widget class="QWidget" name="Tab_Opt">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1242</width>
    <height>642</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QGroupBox" name="groupBox_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Initial Generation</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="2" column="0" colspan="2">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Initial Seed Structures:</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0" colspan="3">
          <widget class="QListWidget" name="list_seeds">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="toolTip">
            <string>Load any initial seed structures here.</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="push_addSeed">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>&amp;Add</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="push_removeSeed">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>&amp;Remove</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Number of initial structures / number of structures to keep in queue for continuous mode.

Default: 20
</string>
          </property>
          <property name="text">
           <string>Initial structures:</string>
          </property>
          <property name="buddy">
           <cstring>spin_numInitial</cstring>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSpinBox" name="spin_numInitial">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Number of initial structures / number of structures to keep in queue for continuous mode.
</string>
          </property>
          <property name="maximum">
           <number>9999</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>Search Parameters</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="label_26">
          <property name="toolTip">
           <string>Number of structures to be considered for genetic operations.

Structures are chosen for genetic operations based on a weighted probability list, such that the lowest enthalpy structures have the greatest probability of being chosen. The number specified here is used to determine how many of the lowest enthalpy structures will have a non-zero probability.

Default: 60</string>
          </property>
          <property name="text">
           <string>Pool size:</string>
          </property>
          <property name="buddy">
           <cstring>spin_popSize</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spin_popSize">
          <property name="toolTip">
           <string>Number of structures to be considered for genetic operations.

Structures are chosen for genetic operations based on a weighted probability list, such that the lowest enthalpy structures have the greatest probability of being chosen. The number specified here is used to determine how many of the lowest enthalpy structures will have a non-zero probability.

Default: 20</string>
          </property>
          <property name="maximum">
           <number>9999</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_genTotal">
          <property name="toolTip">
           <string>How many structures are created for population &gt; 1. This has no effect in continuous mode.

Default: 10</string>
          </property>
          <property name="text">
           <string>Continuous structures:</string>
          </property>
          <property name="buddy">
           <cstring>spin_contStructs</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spin_contStructs">
          <property name="toolTip">
           <string>Number of structures to be kept &quot;in progress&quot; during after the initial generation.

This should depend on the resources available. If a running job limit is set below,

[running job limit] + 1

is a good value here.</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QCheckBox" name="cb_limitRunningJobs">
          <property name="toolTip">
           <string>Check this box to set a limit on the number of jobs that are submitted for a local optimization at any given time.</string>
          </property>
          <property name="text">
           <string>Limit running jobs?</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="spin_runningJobLimit">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Maximum number of simultaneous local optimizations allowed.</string>
          </property>
          <property name="suffix">
           <string> jobs</string>
          </property>
          <property name="prefix">
           <string>Limit to </string>
          </property>
          <property name="value">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>If a job fails </string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spin_failLimit">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="suffix">
             <string> times,</string>
            </property>
            <property name="maximum">
             <number>100</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="combo_failAction">
            <property name="toolTip">
             <string>Either keep trying to optimize, kill, or randomize individuals that fail more than the specified number of times.</string>
            </property>
            <item>
             <property name="text">
              <string>keep trying to optimize.</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>kill the structure.</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>replace with random.</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item row="5" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>If an atom is</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spin_explode">
            <property name="suffix">
             <string> Å</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>away from the cluster:</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="7" column="0" colspan="2">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="combo_explode">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>replace with random.</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>kill the structure.</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="toolTip">
        <string>Experimental: These options are used to fine-tune matching of duplicate structures.</string>
       </property>
       <property name="title">
        <string>Duplicate matching tolerances</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="2">
         <widget class="QDoubleSpinBox" name="spin_tol_enthalpy">
          <property name="toolTip">
           <string>Tolerance for enthalpy comparisons.</string>
          </property>
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="singleStep">
           <double>0.001000000000000</double>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Experimental: These options are used to fine-tune matching of duplicate structures.</string>
          </property>
          <property name="text">
           <string>Enthalpy (eV/atom)</string>
          </property>
          <property name="buddy">
           <cstring>spin_tol_enthalpy</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="2">
         <widget class="QPushButton" name="push_dup_reset">
          <property name="toolTip">
           <string>Manually recheck all structures for duplicates (necessary after changing the above options).</string>
          </property>
          <property name="text">
           <string>Reset duplicates...</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Geometry</string>
          </property>
          <property name="buddy">
           <cstring>spin_tol_geo</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QDoubleSpinBox" name="spin_tol_geo">
          <property name="toolTip">
           <string>Geometry tolerance setting</string>
          </property>
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="maximum">
           <double>100000.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="title">
        <string>Crossover</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="2" column="0">
         <widget class="QLabel" name="label_25">
          <property name="toolTip">
           <string>Percentage of new structures to be created from the heredity operator.

Heredity combines two random spatially coherent slabs from two unique parents to form a single offspring.

Default: 85</string>
          </property>
          <property name="text">
           <string>Percent new crossover:</string>
          </property>
          <property name="buddy">
           <cstring>spin_p_cross</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="spin_p_cross">
          <property name="toolTip">
           <string>Percentage of new structures to be created from the crossover operator.

Default: 15</string>
          </property>
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>15</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_8">
       <property name="title">
        <string>Twist</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Percent new twist:</string>
          </property>
          <property name="buddy">
           <cstring>spin_p_twist</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spin_p_twist">
          <property name="toolTip">
           <string>Percentage of new structures to be created from the crossover operator.

Default: 15</string>
          </property>
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Minimum rotation:</string>
          </property>
          <property name="buddy">
           <cstring>spin_twist_minRot</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spin_twist_minRot">
          <property name="toolTip">
           <string>The smallest rotation allowed during the twist.</string>
          </property>
          <property name="suffix">
           <string>°</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="title">
        <string>Exchange</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QLabel" name="label_29">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Percent new exchange:</string>
          </property>
          <property name="buddy">
           <cstring>spin_p_exch</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_27">
          <property name="toolTip">
           <string>Standard deviation of the random elements of the strain matrix.

Default: 0.07</string>
          </property>
          <property name="text">
           <string>Number of exchanges:</string>
          </property>
          <property name="buddy">
           <cstring>spin_exch_numExch</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spin_p_exch">
          <property name="toolTip">
           <string/>
          </property>
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spin_exch_numExch"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="title">
        <string>Random Walk</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="0" column="0">
         <widget class="QLabel" name="label_31">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Percentage of offspring structures that will be generated from the permutation operator.

The permutation operator will swap two atoms of different type within a single parent. Set to zero if there is only one atom type in the cell!

Default: 5%</string>
          </property>
          <property name="text">
           <string>Percent new:</string>
          </property>
          <property name="buddy">
           <cstring>spin_p_randw</cstring>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_30">
          <property name="toolTip">
           <string>Number of separate swaps in the permutation operator.

Default: 3</string>
          </property>
          <property name="text">
           <string>Displacement range:</string>
          </property>
          <property name="buddy">
           <cstring>spin_randw_minWalk</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spin_p_randw">
          <property name="toolTip">
           <string/>
          </property>
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>35</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Number of walkers:</string>
          </property>
          <property name="buddy">
           <cstring>spin_randw_numWalkers</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spin_randw_numWalkers"/>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QDoubleSpinBox" name="spin_randw_minWalk">
            <property name="suffix">
             <string> Å</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="spin_randw_maxWalk">
            <property name="suffix">
             <string> Å</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_9">
       <property name="title">
        <string>Anisotropic Expansion</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_10">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Percent new:</string>
          </property>
          <property name="buddy">
           <cstring>spin_p_aniso</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spin_p_aniso">
          <property name="suffix">
           <string>%</string>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="spin_aniso_amp"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Amplitude:</string>
          </property>
          <property name="buddy">
           <cstring>spin_aniso_amp</cstring>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_3">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>list_seeds</tabstop>
  <tabstop>push_addSeed</tabstop>
  <tabstop>push_removeSeed</tabstop>
  <tabstop>spin_popSize</tabstop>
  <tabstop>spin_contStructs</tabstop>
  <tabstop>cb_limitRunningJobs</tabstop>
  <tabstop>spin_runningJobLimit</tabstop>
  <tabstop>spin_failLimit</tabstop>
  <tabstop>combo_failAction</tabstop>
  <tabstop>spin_tol_enthalpy</tabstop>
  <tabstop>spin_tol_geo</tabstop>
  <tabstop>push_dup_reset</tabstop>
  <tabstop>spin_p_cross</tabstop>
  <tabstop>spin_p_twist</tabstop>
  <tabstop>spin_twist_minRot</tabstop>
  <tabstop>spin_p_exch</tabstop>
  <tabstop>spin_exch_numExch</tabstop>
  <tabstop>spin_p_randw</tabstop>
  <tabstop>spin_randw_numWalkers</tabstop>
  <tabstop>spin_randw_minWalk</tabstop>
  <tabstop>spin_randw_maxWalk</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cb_limitRunningJobs</sender>
   <signal>toggled(bool)</signal>
   <receiver>spin_runningJobLimit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>618</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>816</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
